projects/aca-shared/src/lib/services/node-permission.service.ts

```diff
@@ -97,4 +97,35 @@ export class NodePermissionService implements NodePermissions {
       return entry.allowableOperations || [];
     }
   }
+
+  private getNodePermissions(node: PermissionSource): PermissionElement[] {
+    let entry: Node | SharedLink;
+    let nodePermissions: any;
+
+    if ('entry' in node) {
+      entry = node.entry;
+    } else {
+      entry = node;
+    }
+
+    if ('permissions' in entry) {
+      nodePermissions = entry.permissions;
+    }
+    return nodePermissions.inherited;
+  }
+
+  hasUserAuthorityOnNode(node: NodeEntry, userGroups: Group[]): boolean {
+    const nodeAuthorities: PermissionElement[] = this.getNodePermissions(node);
+
+    for (const nodeAuth of nodeAuthorities) {
+      for (const userAuth of userGroups) {
+        if (nodeAuth.authorityId === userAuth.id) {
+          if (nodeAuth.name === 'SiteContributor' || nodeAuth.name === 'SiteCollaborator' || nodeAuth.name === 'SiteManager') {
```

Comment:
there is an enum in the JS-API for SiteContributor SiteCollaborator SiteManager better to use it